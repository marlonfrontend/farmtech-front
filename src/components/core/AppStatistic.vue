<script setup lang="ts">
const props = defineProps({
  title: String,
  value: [String, Number],
  bordered: Boolean
})

defineExpose(props)
</script>

<template>
  <div class="text-center" :class="bordered ? 'border-y py-2 sm:border-x sm:border-y-0' : null">
    <div class="font-medium" v-html="title" />
    <span class="font-serif text-primary text-xl" v-html="value === null ? '-' : value" />
  </div>
</template>
