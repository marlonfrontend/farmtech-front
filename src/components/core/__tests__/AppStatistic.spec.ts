import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import AppStatistic from '../AppStatistic.vue'

describe('AppStatistic.vue', () => {
  it('should render component props', () => {
    const props = {
      title: 'Stats Title',
      value: 10
    }
    const wrapper = mount(AppStatistic, {
      propsData: { ...props }
    })
    expect(wrapper.props()).toContain(props)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
