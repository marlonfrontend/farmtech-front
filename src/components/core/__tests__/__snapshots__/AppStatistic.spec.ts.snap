// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppStatistic.vue > should render component props 1`] = `
"<div class=\\"text-center\\">
  <div data-testid=\\"title\\" class=\\"font-medium\\">Hello Word</div>
  <div class=\\"font-serif text-primary text-xl\\"><span data-testid=\\"value\\">10</span></div>
</div>"
`;
